lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -132,6 +170,19 @@ def do_login(self):
         if self.with_tags:
             self.rest_content = self._login_vapi()
 
+        # Get the folder object that will serve as our inventory root folder
+        if self.search_folder:
+            self.root_folder = self.content.searchIndex.FindByInventoryPath('%s/vm/%s' % (self.datacenter, self.search_folder))
+            if not self.root_folder:
+                raise AnsibleError("Specified datacenter + search_folder '%s/vm/%s' were not found on Inventory." %
+                                   (self.datacenter, self.search_folder))
+        elif self.with_folders:
```

Comment:
```suggestion
        elif self.datacenter:
```
This would achieve the same you proposed, I believe. Since with_folders requires the datacenter param anyway, and what is done here is just setting the root folder to the vm folder of the DC.

I see you searched for the DC name and got the vmFolder from it. Is that a more robust approach? Seems like it. I'll use that.

Reply:
@Akasurde  I've pushed this change now, as my suggestion above, and amending the description doc about datacenter param.